src/reflect/scala/reflect/internal/tpe/GlbLubs.scala

```diff
@@ -230,14 +230,17 @@ private[internal] trait GlbLubs {
   def weakLub(tps: List[Type]): Type = (
     if (tps.isEmpty)
       NothingTpe
-    else if (tps forall isNumericValueType)
+    else if (shouldUseNumericLub(tps))
       numericLub(tps)
     else if (tps exists typeHasAnnotations)
       annotationsLub(lub(tps map (_.withoutAnnotations)), tps)
     else
       lub(tps)
   )
 
+  private def shouldUseNumericLub(tps: List[Type]) =
+    tps forall (tp => isNumericValueType(tp.widen))
```

Comment:
I, personally, would expect, in pseudo-scala:

* `weakLub(1.type, 1.type) === 1.type`
* `weakLub(1.type, 2.type) === Int`
* `weakLub(1.type, Int) === Int`
* `weakLub(1.type, 1L.type) === Long`
* `weakLub(1.type, Long) === Long`
* `weakLub(1.type, ().type) === AnyVal`
* `weakLub(1.type, "".type) === Any`
* `weakLub(1.type, String) === Any`.

and various similar results. Moreover, I'd expect those to be consistent over type-level eta-expansion; given `type I1 = 1`, `weakLub(I1, Long) === Long` &c.

It seems to me that `.dealias.deconst` in `isNumericValueType` would produce that behaviour. I don't see anything in [the updated section of the spec](https://github.com/scala/scala/pull/5310/files#diff-c067e953d98d0b8fb993aeae8bdb0bd4) which would indicate otherwise.